rollup-interface/src/state_machine/zk/mod.rs

```diff
@@ -61,20 +73,28 @@ pub trait ValidityCondition: Serialize + DeserializeOwned {
 /// if and only if the condition `validity_condition` is satisfied.
 ///
 /// The period of time covered by a state transition proof may be a single slot, or a range of slots on the DA layer.
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct StateTransition<C> {
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct StateTransition<C, Address> {
```

Comment:
Is `Address` parameter can be anything, or do we want to trait bound it to `AddressTrait`?